### DISCUSSION TITLE
How to use sqlite or sequelize or typeorm on react side?

### DISCUSSION THREAD

**Discussion by leopucci** [MPMcIntyre/ReacTron#14] [Q&A] [answered]
Hi, does someone knows how to properly access sqlite or typeorm or sequelize inside react?

**Comment by mpmcintyre** [comment, marked as answer]
Hi @leopucci 

Generally it is not considered good practice to allow node modules to be accessed on the renderer (react) side as it allows for [cross-site-scripting](https://owasp.org/www-community/attacks/xss/) (XSS) attacks, however, it is possible by allowing it on the window.

You would do this by changing the settings on the webPreferences as such:

```
  webPreferences: {
    preload: preloadFilePath,
    //!security
    nodeIntegration: true, //unsecure
    contextIsolation: false, // unsecure
    enableRemoteModule: true, // unsecure
  },
```

so a new window would look like this:

```
const win = new BrowserWindow({
  width: 800,
  height: 600,
  webPreferences: {
    preload: preloadFilePath,
    //!security
    nodeIntegration: true,
    contextIsolation: false,
    enableRemoteModule: true,
  },
});

// * Load file
win.loadFile(mainHtmlFilePath);
```


You should then be able to access any node module inside your renderer (react) process

kind regards

**Reply by leopucci** [reply]
Thanks!
